File: Invoice_Fraud_Detection/src/main/java/com/invoicefraud/service/VendorService.java
package com.invoicefraud.service;

import java.util.List;

import com.invoicefraud.dto.VendorRequestDTO;
import com.invoicefraud.dto.VendorResponseDTO;

public interface VendorService {

	VendorResponseDTO createVendor(VendorRequestDTO requestDTO);

	VendorResponseDTO getVendorById(Long id);

	List<VendorResponseDTO> getAllVendors();

	VendorResponseDTO updateVendor(Long id, VendorRequestDTO requestDTO);

	void deleteVendor(Long id);
}

File: frontend/src/services/vendorService.js
import api from '../api/axios';

const vendorService = {
  getAllVendors: () => api.get('/vendors'),
  
  getVendorById: (id) => api.get(`/vendors/${id}`),
  
  createVendor: (vendorData) => api.post('/vendors', vendorData),
  
  updateVendor: (id, vendorData) => api.put(`/vendors/${id}`, vendorData),
  
  deleteVendor: (id) => api.delete(`/vendors/${id}`),
};

export default vendorService;
